README.md

```diff
@@ -1,345 +1,451 @@
-communicate between them. This is suitable for many micro service architectures.
+This is a framework to use [RabbitMQ] as a client/server RPC setup togheter with
+the [Go amqp] implementation. The framework can manage a fully funcitonal
+message queue setup with reconnects, disconnects, graceful shutdown and other
+stability mechanisms. By providing this RabbitMQ can be used as transport and
+service discovery to quickly get up and running with a micro service
+architecture.
+
+Since the framework handles the majority of the interface with RabbitMQ and the
+Go library the user does not need to know details about those systems. However,
+since a few interfaces exposes the Go package types and that the nomenclature is
+unique for RabbitMQ some prior experience is preferred.
 
-We assume that the user has some knowledge about RabbitMQ and preferrably the Go
-package since a few of the types are exposed to the end user. However, for
-simple setups there's no need for understanding of any of this.
+## Project status
 
-## Components
+This project has been used in production since october 2018 handling millions of
+requests both as server and client. The implementation is to be seen as stable
+altough until a proper v1.0 release tag is set the API may change.
 
-This framework consists of a client and a server with related beloning
-components such as request, responses, connections and other handy types.
+## Server
 
-### Server
+The server is inspired by the http library where the user maps a [RabbitMQ
+binding] to a handler function. A response writer is passed to the handler which
+may be used as an `io.Writer` to write the response.
 
-The server is designed to be a *plug-and-play* component with handlers attached
-to endpoints for amqp messages. All you need to do to start consuming
-messages published to `routing_key` looks like this:
+This is an an example of how to get up and running with a server responding to
+all messages published to the given routing key.
 
 ```go
-s := NewServer("amqp://guest:guest@localhost:5672")
+server := NewServer("amqp://guest:guest@localhost:5672")
 
-s.Bind(DirectBinding("routing_key", func(c context.Context, rw *ResponseWriter d *amqp.Delivery) {
+server.Bind(DirectBinding("routing_key", func(c context.Context, rw *ResponseWriter d *amqp.Delivery) {
     // Print what the body and header was
     fmt.Println(d.Body, d.Headers)
 
-    // Add a response to the client
+    // Add a response to the publisher
     fmt.Fprint(rw, "Handled")
 }))
 
-s.ListenAndServe()
+server.ListenAndServe()
 ```
 
-This example will use the default exchange for direct bindings (`direct`) and
-use the routing key provided as queue name. It's also possible to specify other
-kind of exchanges such as topic or fanout by using the `HandlerBinding` type.
-This package already supports direct, fanout, topic and header.
+The example above uses a `DirectBinding` but all the supported bindings is
```

Comment:
is → are

Reply:
🤦‍♂️ - Will never accept that I'm till having issues with is/are